hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -927,6 +923,44 @@ private static File urlToFile(@Nonnull URL url) {
         return resourceConfigs;
     }
 
+    /**
+     * Registers given serializer for a given class for the scope of the job.
+     * Both will be accessible to all the code attached to the underlying
+     * pipeline or DAG, but not to any other code. (An important example is
+     * the {@code IMap} data source, which can instantiate only the classes
+     * from the Jet instance's classpath.)
+     *
+     * Serializer must have no-arg constructor.
+     *
+     * @implNote Backing storage for this method is an {@link IMap} with a
+     * default backup count of 1. When adding big files as a resource, size
+     * the cluster accordingly in terms of memory, since each file will have 2
+     * copies inside the cluster(primary + backup replica).
+     *
+     * @return {@code this} instance for fluent API
+     */
+    @Nonnull
+    public <T, S extends StreamSerializer<?>> JobConfig addSerializer(Class<T> clazz, Class<S> serializerClass) {
```

Comment:
should we not offer a more convenient interface here? like Serializer<T> or similar?

Reply:
agreed with @vbekiaris that `StreamSerializer` should be enough, `ByteArraySerializer` is more of a private thingy